src/components/Admin/Approval/Approval.js

```diff
@@ -0,0 +1,150 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+import ItemSelector from '../ItemSelector';
+import DataController from '../utils/DataController';
+
+import './Approval.css';
+
+class ImageApproval extends Component {
+  state = {
+    loading: true,
+    images: [],
+    storeList: [{name: 'unknown', id: 'unknown'}]
+  };
+
+  getImages = async () => {
+    return this.dataController
+      .getImages(false, 10)
+      .then(images => this.setState({images}));
+  };
+
+  lastImageTimestamp = () => {
+    return this.state.images.reduce(
+      (highest, current) => Math.max(highest, current.timestamp),
+      0
+    );
+  };
+
+  nextImage = index => {
+    return this.dataController
+      .getImages(false, 1, this.lastImageTimestamp())
+      .then(images => {
+        this.setState(prevState => {
+          if (images.length < 1) {
+            prevState.images.splice(index, 1);
+          } else {
+            prevState.images[index] = images[0];
+          }
+          return prevState;
+        });
+      });
+  };
+
+  trustImage = event => {
+    const index = event.target.dataset.index;
+    this.dataController
+      .trustImage(event.target.dataset.id)
+      .then(() => this.nextImage(index));
+  };
+
+  setAsUnknown = event => {
+    const index = event.target.dataset.index;
+    this.dataController.changeImageLabel(event.target.dataset.id, 'unknown');
+    this.dataController
+      .trustImage(event.target.dataset.id)
+      .then(() => this.nextImage(index));
+  };
+
+  deleteImage = event => {
+    const index = event.target.dataset.index;
+    this.dataController
+      .deleteImage(event.target.dataset.id)
+      .then(() => this.nextImage(index));
+  };
+
+  changeCategory = (id, newCategory) => {
+    this.dataController.changeImageLabel(id, newCategory);
+  };
+
+  back = () => {
+    this.props.history.replace('/admin');
+  };
+
+  componentDidMount() {
+    this.dataController = new DataController();
+    this.dataController.getStoreList().then(storeList => {
+      this.setState({
+        storeList: Object.values({
+          ...storeList
+        })
+      });
+      this.getImages().then(() => this.setState({loading: false}));
+    });
+  }
+
+  render() {
+    if (this.state.loading) return <div>Loading...</div>;
+    if (this.state.images === [])
+      return <div>No untrusted images available</div>;
+
+    return (
+      <div className="preview">
+        <div>
```

Comment:
do you need this div?

Reply:
Makes it look nicer by rendering it in a block, puts everything after below it.

Reply:
and all the other divs that do nothing?